### DISCUSSION TITLE
RadioLib Frequency hopping?

### DISCUSSION THREAD

**Discussion by DavidStacer** [jgromes/RadioLib#894] [Q&A]
Does RadioLib implement  frequency hopping as required for some locations and frequencies?  

Requirement FCC  15.247(a)

**Comment by jgromes** [comment]
No idea what you're asking about, sorry. It's up to the user to ensure compliance with their local regulations - as per the license.

**Reply by DavidStacer** [reply]
In the USA under FCC ISM 900Mhz, you can't just transmit at one frequency for more than 0.4 of a second, so most systems implement a method of hopping around a group of frequencies/channels.  

https://www.govinfo.gov/content/pkg/CFR-2013-title47-vol1/pdf/CFR-2013-title47-vol1-sec15-247.pdf

**Reply by jgromes** [reply]
Again, that's up to the user. RadioLib supports whatever frequency bands are supported by the hardware.

**Comment by alx-uta** [comment]
@DavidStacer I'm not sure if you already fixed this, but I did this to receive data from a second frequency.

```
void rxHopping() {
    current_time = millis();

    if ((
        current_time - last_switch_time
    ) >= SX1280_HOPPING_INTERNAL_MILLIS and current_frequency!=SX1280_TX_FREQUENCY) {

        // Move to the External TX Frequency
        current_frequency = SX1280_TX_FREQUENCY;
        radio.setFrequency(current_frequency);

        // Update last_switch_time to the current time to mark the last switch time
        last_switch_time = current_time;
    }

    if ((
        current_time - last_switch_time
    ) >= SX1280_TX_FREQUENCY_AVAILABILITY_MILLIS and current_frequency!=SX1280_FREQUENCY) {

        // Move to the TX Frequency
        current_frequency = SX1280_FREQUENCY;
        radio.setFrequency(current_frequency);

        // Update last_switch_time to the current time to mark the last switch time
        last_switch_time = current_time;
    }
}
```

In your case, I think you'd need to use a default frequency, and after you'll receive the first package to switch on a different one.
